traitsui/testing/ui_tester.py

```diff
@@ -0,0 +1,264 @@
+
+from contextlib import contextmanager
+
+from pyface.gui import GUI
+
+from traitsui.testing.simulation import (
+    get_editor_value, set_editor_value, REGISTRY
+)
+from traitsui.tests._tools import store_exceptions_on_all_threads
+
+
+class UITester:
```

Comment:
Yes: The `click` method defined in `BaseSimulator` can be implemented for `ButtonEditor` for performing the same effect of `click_button`.
The `press_ok_button` in `_tools.py` currently just presses the ONE button it finds without checking if it is the OK button or if there are other buttons. That probably works for the few tests that use it, but moving that over here as is would be a bad idea. I think I need to think harder about that one...

Reply:
`click_button` can also be used with `set_editor` and any other editors that save a reference to the button (as the control of the button can be passed directly). What I'm thinking about is keeping those functions as they are but maybe making them static methods of `UITester` so that they are exposed publicly, and adding processing of GUI events as required. I don't think they need to go through any simulator as they don't really need to know anything about a specific editor.

Again, I'm not really thinking about using this outside testing, so there might be an argument that my suggestion would make these functions unavailable for other uses.

Reply:
I see.

We could decide to make `traitsui.tests._tools.click_button` public, but I prefer keeping it out of `UITester`. 

The purpose of `UITester` is to promote and support test code that is free of implementation details of an editor.  I am afraid in order to use `click_button`, one still needs to know where a widget is in an editor. For example, the use of `click_button` for testing `SetEditor` requires one to access GUI widgets like `_use` and `_unuse` (which are QPushButton when Qt is used).

Again there is a question of how common is it to test pressing up-down-left-right buttons in a set editor in downstream projects. If it is common, then we might be better off rolling out a completely separate tester for driving interactions for a more complicated editor.